File: Fase 2/ServidorRevistas/src/main/java/objetos/Suscripcion.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package objetos;

import java.util.Date;

/**
 *
 * @author fernanrod
 */
public class Suscripcion {
    private Integer id;
    private String usuario;
    private Integer revista;
    private Date fechaSuscripcion;
    private Date fechaCancelacion;
    
    public Suscripcion(Integer id, String usuario, Integer revista, Date fechaSuscripcion, Date fechaCancelacion) {
        this.id = id;
        this.usuario = usuario;
        this.revista = revista;
        this.fechaSuscripcion = fechaSuscripcion;
        this.fechaCancelacion = fechaCancelacion;
    }
    
    public Integer getId() {
        return id;
    }
    public void setId(Integer id) {
        this.id = id;
    }
    public String getUsuario() {
        return usuario;
    }
    public void setUsuario(String usuario) {
        this.usuario = usuario;
    }
    public Integer getRevista() {
        return revista;
    }
    public void setRevista(Integer revista) {
        this.revista = revista;
    }
    public Date getFechaSuscripcion() {
        return fechaSuscripcion;
    }
    public void setFechaSuscripcion(Date fechaSuscripcion) {
        this.fechaSuscripcion = fechaSuscripcion;
    }
    public Date getFechaCancelacion() {
        return fechaCancelacion;
    }
    public void setFechaCancelacion(Date fechaCancelacion) {
        this.fechaCancelacion = fechaCancelacion;
    }
}

File: Fase 2/revistas-app/src/app/objetos/base/Suscripcion.ts
export class Suscripcion {
    private _id: number;
    private _usuario: string;
    private _revista: number;
    private _fechaSuscripcion: Date;
    private _fechaCancelacion: Date|null;
    
    constructor(id: number, usuario: string, revista: number, fechaSuscripcion: Date, fechaCancelacion: Date|null) {
        this._id = id;
        this._usuario = usuario;
        this._revista = revista;
        this._fechaSuscripcion = fechaSuscripcion;
        this._fechaCancelacion = fechaCancelacion;
    }
    
    public get id(): number {
        return this._id;
    }
    public get usuario(): string {
        return this._usuario;
    }
    public get revista(): number {
        return this._revista;
    }
    public get fechaSuscripcion(): Date {
        return this._fechaSuscripcion;
    }
    public get fechaCancelacion(): Date|null {
        return this._fechaCancelacion;
    }
}